akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala

```diff
@@ -79,9 +82,23 @@ private[akka] final case class EventSourcedBehaviorImpl[Command, Event, State](
     val actualSignalHandler: PartialFunction[Signal, Unit] = signalHandler.orElse {
       // default signal handler is always the fallback
       case SnapshotCompleted(meta: SnapshotMetadata) ⇒
-        ctx.log.debug("Save snapshot successful, snapshot metadata: [{}]", meta)
+        ctx.log.debug("Save snapshot successful, snapshot metadata [{}]", meta)
       case SnapshotFailed(meta, failure) ⇒
-        ctx.log.error(failure, "Save snapshot failed, snapshot metadata: [{}]", meta)
+        ctx.log.error(failure, "Save snapshot failed, snapshot metadata [{}]", meta)
+      case DeleteSnapshotsCompleted(DeletionTarget.Individual(meta)) =>
+        ctx.log.debug(s"Persistent snapshot [{}] deleted successfully.", meta)
+      case DeleteSnapshotsCompleted(DeletionTarget.Criteria(criteria)) =>
+        ctx.log.debug(s"Persistent snapshots given criteria [{}] deleted successfully.", criteria)
+      case DeleteSnapshotsFailed(DeletionTarget.Individual(meta), failure) =>
+        ctx.log.warning(failure, "Failed to delete snapshot with meta [{}].", meta)
+      case DeleteSnapshotsFailed(DeletionTarget.Criteria(criteria), failure) =>
+        ctx.log.warning("Failed to delete snapshots given criteria [{}] due to [{}].", criteria, failure.getMessage)
```

Comment:
my suggestion was to use the full `failure` as parameter in those instead of `failure.getMessage`. Then `toString` of the Throwable will  include the class name as you originally wanted to include (and that is useful)

Reply:
Updated.